common/djangoapps/util/enterprise_helpers.py

```diff
@@ -120,3 +124,62 @@ def insert_enterprise_pipeline_elements(pipeline):
 
     for index, element in enumerate(additional_elements):
         pipeline.insert(insert_point + index, element)
+
+
+def get_enterprise_customer_logo_url(request):
+    """
+    Client API operation adapter/wrapper
+    """
+
+    if not enterprise_enabled():
+        return None
+
+    parameter = get_enterprise_branding_filter_param(request)
+    if not parameter:
+        return None
+
+    provider_id = parameter.get('provider_id', None)
+    ec_uuid = parameter.get('ec_uuid', None)
+
+    if provider_id:
+        branding_info = enterprise_api.get_enterprise_branding_info_by_provider_id(provider_id=provider_id)
+    elif ec_uuid:
+        branding_info = enterprise_api.get_enterprise_branding_info_by_ec_uuid(ec_uuid=ec_uuid)
+
+    logo_url = None
+    if branding_info and branding_info.logo:
+        logo_url = branding_info.logo.url
+
+    return logo_url
+
+
+def set_enterprise_branding_filter_param(request, provider_id=None):
+    """
+    Setting 'EC_BRANDING_FILTER_PARAM' in session. 'EC_BRANDING_FILTER_PARAM' either be provider_id or ec_uuid.
+    e.g. {provider_id: 'xyz'} or {ec_src: enterprise_customer_uuid}
+    """
+    ec_uuid = request.GET.get('ec_src', None)
+    if EC_BRANDING_FILTER_PARAM in request.session and (provider_id or ec_uuid):
```

Comment:
I understand the case you're trying for, but the extra logic is unnecessary. Here's whats going on:

```python
def set_value(my_dict, a=None, b=None):
    if a or b:
        del my_dict['value']

    if a:
        my_dict['value'] = a
    elif b:
        my_dict['value'] = b
```

The end result is that if either `a` or `b` is set, then `my_dict['value']` will be set to one of those values. It is unnecessary to first delete the `value` key for `my_dict`, because when setting a value, we do not retain the old value.

In other words, if `my_dict == {'a': {'value': 1}}`, then doing `my_dict['a'] = 'another value'` will result in `my_dict == {'a': 'another_value'}`, without the necessity to first do `del my_dict['a']`. What I am asking is that you convert the above block of code to the following:

```python
def set_value(my_dict, a=None, b=None):
    if a:
        my_dict['value'] = a
    elif b:
        my_dict['value'] = b
```

There is no set of conditions in which we will reach the block of code that deletes the existing value from the session dictionary in which we will not also entirely replace that subset of the session dictionary with a new dictionary that does not contain any trace of the existing value.

To demonstrate this, try replacing the `del request.session[ENTERPRISE_CUSTOMER_BRANDING_OVERRIDE_DETAILS]` line with something like `request.session[ENTERPRISE_CUSTOMER_BRANDING_OVERRIDE_DETAILS] = 'This is a string that does not work at all! It cannot store session details! But it causes no errors because it is not used, ever!'`. You can then observe that the manipulations you make in that block have no bearing on the end result.